apis/components/v1alpha1/datasciencepipelines_types.go

```diff
@@ -53,7 +53,8 @@ type DataSciencePipelinesSpec struct {
 }
 
 type DataSciencePipelinesCommonSpec struct {
-	common.DevFlagsSpec `json:",inline"`
+	common.DevFlagsSpec    `json:",inline"`
+	ModelRegistryNamespace string `json:"modelRegistryNamespace,omitempty"`
```

Comment:
what happen if the model registry is not enabled ? maybe it should follow the same pattern as https://github.com/opendatahub-io/opendatahub-operator/pull/1682 so we would need to pass both the management state and the namespace ?

Reply:
i am not sure if DSPO has any checks on the existance of ModelReg. from ticket requirement it does not state this.
thats why i have "do not care if ModelReg is enabled or not, it always has NS value set in DSC by default" in the descripton.
if they need the managementstate of ModelReg, i can add this into PR

@mprahl ^

Reply:
@lburgazzoli good point! We don't need the management state since the InstructLab pipeline will accept the Model Registry name as input and check if it can connect to it. The main goal here is to avoid the user needing to know the namespace that the Model Registry is deployed in.

Reply:
@zdtsw would it be possible to have a sha256 hash of the params.env file contents as an environment variable value on the DSPO so that way any time the params.env file changes, the DSPO will restart and look at the new values?

This would make it so that we don't need a new field on the CRD and the DSPO doesn't need custom logic around this.

Reply:
i think you can do it from DSPO directly, 
simliar to https://github.com/opendatahub-io/odh-model-controller/blob/incubating/config/manager/manager.yaml#L103 with https://github.com/opendatahub-io/odh-model-controller/blob/incubating/config/base/params.env#L8

Reply:
@zdtsw perhaps I'm misunderstanding but I thought params.env was provided as a ConfigMap to DSPO. If it's not, how is it provided?

Reply:
The process is:
- the `params.env` is translated to a `configmap` https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/base/kustomization.yaml#L15-L18
- the values from `configmap` translated to some `variables` https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/base/kustomization.yaml#L19
- the `variables` then are then referenced in some other `manifests`, i.e, https://github.com/opendatahub-io/data-science-pipelines-operator/blob/3f867c3835878d98f37801c0b9520f91d845a129/config/manager/manager.yaml#L72-L73

So, if the model registry namespace, ends up being set to an env var, then it would trigger the restart of the pod

Reply:
Thanks for the detailed explanation! So if it's set in params.env, then the DSPO manifests can set it as an environment variable (e.g. `DSPO_MODELREGISTRYNAMESPACE`) which will restart the DSPO when the value is updated. So that works for me!

Reply:
> 
> So, if the model registry namespace, ends up being set to an env var, then it would trigger the restart of the pod

Should not we get rid of vars in favour of replacements?

https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/vars/

Reply:
@ykaliuta could you expand a bit on what you mean?

Reply:
> @ykaliuta could you expand a bit on what you mean?

As the link says
```
The vars field was deprecated in v5.0.0. This field will never be removed from the kustomize.config.k8s.io/v1beta1 Kustomization API, but it will not be included in the kustomize.config.k8s.io/v1 Kustomization API. When Kustomization v1 is available, we will announce the deprecation of the v1beta1 version. There will be at least two releases between deprecation and removal of Kustomization v1beta1 support from the kustomize CLI, and removal itself will happen in a future major version bump.

Please try to migrate to the the [replacements](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/replacements) field.
```

Reply:
It does not affect the general logic where `env` must be changed to restart the container.